New: swap

ranneyd opened the pull request:
Ramda 0.29.0 introduced a [simple swap function](https://ramdajs.com/docs/#swap). Here's the typing for it. Full disclosure: I copy/pasted the code from `adjust.d.ts` since the function signatures are similar.

Harris-Miller commented:
Nice. Just do two things for me:
* reverse the overload order (see [reduce.d.ts ](https://github.com/ramda/types/blob/develop/types/reduce.d.ts)for an example of the correct ordering)
* add tests

ranneyd commented:
@Harris-Miller thanks for the quick response! On it.

ranneyd commented:
@Harris-Miller could you check again now?

ranneyd commented:
@Harris-Miller can you take another look?

Note: I had some lint errors. I fixed those and also updated `CONTRIBUTING.md` to call out running `npm run lint` during testing. 😄

lax4mike commented on types/swap.d.ts:
```diff
@@ -0,0 +1,58 @@
+  indexB: number
+): {
+  // swap(__, indexB)(indexA)(list)
+  (indexA: number): <T>(list: readonly T[]) => T[];
```
I'm curious about the `readonly` keyword here.  Is this just preventing the definition of `swap` from mutating the list argument?  I guess the implementation of `swap` is already written in the other repo and these definitions do not restrict that implementation.  

Does `readonly` do anything else that restricts the consumer or type inference in any way?  It probably can't hurt to have these here, but just wanted to see if I was missing something else.

lax4mike replied:
ahh, that makes sense.   It does help the consumer in that this function now accepts `readonly` arrays.  Thanks for the explanation!

Harris-Miller replied:
[playground example](https://tsplay.dev/WzxjeN)

lax4mike reviewed (approved):
Look good!
